src/test/java/org/opensearch/performanceanalyzer/rest/QueryRcaRequestHandlerTest.java

```diff
@@ -0,0 +1,144 @@
+    private QueryRcaRequestHandler handler;
+    private AppContext appContext;
+    private final String queryPrefix = "http://localhost:9600/_plugins/_performanceanalyzer/rca";
+
+    private void setClusterManagerContext(boolean isClusterManager) {
+        ClusterDetailsEventProcessor clusterDetailsEventProcessor =
+                new ClusterDetailsEventProcessor();
+        clusterDetailsEventProcessor.setNodesDetails(
+                Collections.singletonList(
+                        new ClusterDetailsEventProcessor.NodeDetails(
+                                isClusterManager
+                                        ? AllMetrics.NodeRole.ELECTED_CLUSTER_MANAGER
+                                        : AllMetrics.NodeRole.DATA,
+                                "test_node",
+                                "127.0.0.1",
+                                isClusterManager)));
+        appContext.setClusterDetailsEventProcessor(clusterDetailsEventProcessor);
+    }
+
+    private HttpExchange sendQuery(String query, String requestMethod, OutputStream os)
+            throws Exception {
+        HttpExchange exchange = Mockito.mock(HttpExchange.class);
+        Mockito.when(exchange.getResponseBody()).thenReturn(os != null ? os : System.out);
+        Mockito.when(exchange.getRequestMethod()).thenReturn(requestMethod);
+        Headers responseHeaders = Mockito.mock(Headers.class);
+        Mockito.when(exchange.getResponseHeaders()).thenReturn(responseHeaders);
+        Mockito.when(exchange.getRequestURI()).thenReturn(new URI(query));
+        handler.handle(exchange);
+        return exchange;
+    }
+
+    @Before
+    public void setUp() {
+        appContext = new AppContext();
+        handler = new QueryRcaRequestHandler(appContext);
+        setClusterManagerContext(false);
+        Stats.getInstance().getConnectedComponents(); // Initializes muted graph nodes
+    }
+
+    @After
+    public void undo() {
+        Stats.getInstance().getConnectedComponents();
+    }
+
+    @Test
+    public void testInvalidNodeRole() throws Exception {
+        HttpExchange exchange = sendQuery(queryPrefix + "?name=HotShardClusterRca", "GET", null);
+        Mockito.verify(exchange)
+                .sendResponseHeaders(
+                        ArgumentMatchers.eq(HttpURLConnection.HTTP_BAD_REQUEST),
+                        ArgumentMatchers.anyLong());
+    }
+
+    @Test
+    public void testValidNodeRole() throws Exception {
+        setClusterManagerContext(true);
+        HttpExchange exchange = sendQuery(queryPrefix + "?name=HotShardClusterRca", "GET", null);
+        Mockito.verify(exchange)
+                .sendResponseHeaders(
+                        ArgumentMatchers.eq(HttpURLConnection.HTTP_OK), ArgumentMatchers.anyLong());
+    }
+
+    @Test
+    public void testBadRequestMethod() throws Exception {
+        HttpExchange exchange = sendQuery(queryPrefix + "?name=HotShardClusterRca", "PUT", null);
```

Comment:
Why is HTTP Verb `PUT` here instead of `GET` ?

Reply:
Also, more general test (this one purely for the sake of coverage - illegal request method).